Helper/AnalyticsHelper.php

```diff
@@ -58,25 +49,17 @@ class AnalyticsHelper
     protected $region;
 
     /**
-     * @param AlgoliaHelper $algoliaHelper
      * @param ConfigHelper $configHelper
      * @param IndexEntityDataProvider $entityHelper
      * @param Logger $logger
-     * @param string $region
+     * @param ResolverInterface $localeResolver
      */
     public function __construct(
-        AlgoliaHelper $algoliaHelper,
-        ConfigHelper $configHelper,
-        IndexEntityDataProvider $entityHelper,
-        Logger $logger,
-        string $region = 'us'
+        private ConfigHelper            $configHelper,
+        private IndexEntityDataProvider $entityHelper,
+        private Logger                  $logger,
+        private ResolverInterface       $localeResolver
```

Comment:
@mrahman3177 Rather than do a straight cherry pick here, can we sift out the property promotion (stick with old style constructor code) to avoid compatibility issues with PHP 7?